games/tic_tac_toe.c

```diff
@@ -80,6 +82,27 @@ int main()
  *
  * @returns None
  */
+int TakeInput(){
+    
+    // Input player's choice of placing 'x' or 'o' in variable x 
+    char x[100];
+    fgets(x, 100, stdin);
+
+    // to check the length of input!
+    int i;
+    for (i = 0; x[i] != '\0'; i++);   
+
+    // if length of input=1 (i.e i=2) and input value is a digit then its a valid input.
+    if(i==2 && x[0]-'0'>=1 && x[0]-'0'<=9  ){   
+        int m=x[0]-'0';
+        return m;
+    }
+    else{
+        printf("(Enter a valid number between [1 - 9]) ");
+        return TakeInput();
+    }
+}
+
 void singlemode()
```

Comment:
the documentation for this function was initially from line 78 to 82 and is misplaced.